### PR TITLE
Docs redundancy

### PR DESCRIPTION
**Pull request by apeltzer**
This should make some points mentioned in https://github.com/nf-core/tools/issues/97 more clear.

I was wrong with the redundancy I think: One document describes more in detail how you can add your own profile in here, whereas the other one describes how to use Docker/Singularity with the specific pipeline, which is something different. 

However, I took the time to update the docs a bit, adding in some more up2data ways how we normally run testcases now (adding a test.config profile for example) and removing the `./tests` directory.

### CODE REVIEW

File: nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ singularity {
   enabled = true
 }
 process {
-  container = "docker://$wf_container"
+  container = "shub://$wf_container"
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md]: `$wf_container` doesn't exist any more - should be `params.container`. Or in fact, I think we are moving in the direction of just hardcoding this as it shouldn't change much.

File: nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md [hunk 2]
```diff
@@ -110,12 +110,9 @@ bash Miniconda3-latest-Linux-x86_64.sh
 
 #### 2) Add the bioconda conda channel (and others)
 ```bash
-conda config --add channels anaconda
+conda config --add channels default
 conda config --add channels conda-forge
-conda config --add channels defaults
-conda config --add channels r
 conda config --add channels bioconda
-conda config --add channels salilab
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md]: hah, where did this creep in from? 😆

File: nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/local.md [hunk 3]
```diff
@@ -25,18 +25,18 @@ The public docker images are tagged with the same version numbers as the code, w
 ## Singularity image
 Many HPC environments are not able to run Docker due to security issues. [Singularity](http://singularity.lbl.gov/) is a tool designed to run on such HPC systems which is very similar to Docker. Even better, it can use create images directly from dockerhub.
 
-To use the singularity image for a single run, use `-with-singularity 'docker://{{ cookiecutter.pipeline_slug }}'`. This will download the docker container from dockerhub and create a singularity image for you dynamically.
+To use the singularity image for a single run, use `-with-singularity 'shub://{{ cookiecutter.pipeline_slug }}'`. This will download the docker container from dockerhub and create a singularity image for you dynamically.
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/local.md]: Should be `nf-core/{{ cookiecutter.pipeline_slug }}`

File: nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/local.md [hunk 4]
```diff
@@ -25,18 +25,18 @@ The public docker images are tagged with the same version numbers as the code, w
...
 
 If you intend to run the pipeline offline, nextflow will not be able to automatically download the singularity image for you. Instead, you'll have to do this yourself manually first, transfer the image file and then point to that.
 
 First, pull the image file where you have an internet connection:
 
 ```bash
-singularity pull --name {{ cookiecutter.pipeline_slug }}.img docker://{{ cookiecutter.pipeline_slug }}
+singularity pull --name {{ cookiecutter.pipeline_slug }}.simg shub://{{ cookiecutter.pipeline_slug }}
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/local.md]: `nf-core/{{ cookiecutter.pipeline_slug }}`

File: nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/local.md [hunk 5]
```diff
@@ -25,18 +25,18 @@ The public docker images are tagged with the same version numbers as the code, w
...
 ```
 
 Then transfer this file and run the pipeline with this path:
 
 ```bash
-nextflow run /path/to/{{ cookiecutter.pipeline_name }} -with-singularity /path/to/{{ cookiecutter.pipeline_slug }}.img
+nextflow run /path/to/{{ cookiecutter.pipeline_name }} -with-singularity /path/to/{{ cookiecutter.pipeline_slug }}.simg
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/local.md]: `nf-core-{{ cookiecutter.pipeline_slug }}`

**Comment by apeltzer** [comment]
Will look after this now :-)

File: nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md [hunk 6]
```diff
@@ -58,18 +58,22 @@ docker {
...
 
+
 To specify singularity usage in your pipeline config file, add the following:
 
 ```nextflow
 singularity {
   enabled = true
 }
+process {
+  container = "shub://$wf_container"
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md]: This is out of date now (we don't define `$wf_container` these days). Should be:

```nextflow
singularity.enabled = true
process.container = {"shub://${params.container.replace('nfcore', 'nf-core')}"}
```

Comment (ewels) [reply nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md]: Or maybe better still:

```
singularity.enabled = true
process.container = params.container
```

Comment (apeltzer) [reply nf_core/pipeline-template/{{cookiecutter.pipeline_slug}}/docs/configuration/adding_your_own.md]: This is still not in my mind :confused:

**Pull request merged**